server/src/main/java/io/crate/metadata/Reference.java

```diff
@@ -69,8 +78,8 @@ public Reference(StreamInput in) throws IOException {
         type = DataTypes.fromStream(in);
         granularity = RowGranularity.fromStream(in);
 
-        columnPolicy = ColumnPolicy.values()[in.readVInt()];
-        indexType = IndexType.values()[in.readVInt()];
+        columnPolicy = ColumnPolicy.VALUES.get(in.readVInt());
+        indexType = IndexType.fromStream(in);
```

Comment:
Minor: Can we use this.VALUES.get(in.readVInt()); just to make it similar to line above and to RowGranularity?

Reply:
We usually follow the `fromStream` static factory pattern  (See `RowGranularity` further above or `SymbolType`, which does the same). The reason I didn't use it for `ColumnPolicy` is because it is part of the sql-parser module and there we don't have the `StreamInput` dependency.

Reply:
I see, thanks for clarification